### PR TITLE
AIP-44 Initialize methods map for Internal API RPC endpoint in the method

### PR DESCRIPTION
**Pull request by mhenc**
To avoid potential cyclic dependencies

### CODE REVIEW

File: airflow/api_internal/endpoints/rpc_api_endpoint.py [hunk 1]
```diff
@@ -33,11 +32,17 @@ def _build_methods_map(list) -> dict:
     return {f"{func.__module__}.{func.__name__}": func for func in list}
 
 
-METHODS_MAP = _build_methods_map(
-    [
-        DagFileProcessor.update_import_errors,
-    ]
-)
+def _initialize_map() -> dict:
```
Comment (uranusjr) [inline airflow/api_internal/endpoints/rpc_api_endpoint.py]: I think we should inline `_build_methods_map` into this.

Comment (mhenc) [reply airflow/api_internal/endpoints/rpc_api_endpoint.py]: Sure.

**Comment by mhenc** [comment]
cc: @vincbeck

**Review by vincbeck — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]
I pushed a change to further delay init (by using lru_cache instead of a global variable)

**Comment by vincbeck** [comment]
Unit tests need to be updated as well (see failing tests)

**Review by vincbeck — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Comment by mhenc** [comment]
cc: @potiuk @uranusjr  Can you merge this please?

**Review by potiuk — APPROVED** [review]

**Pull request merged**